### ISSUE TITLE
Cancelling hint mode with Esc doesn't work most of the time

### ISSUE THREAD

**Issue by maxigaz** [gdh1995/vimium-c#810]
**Describe the bug**

After upgrading to Vimium C v1.99.9, cancelling hint mode with `Esc` doesn't work most of the time. To temporarily make it work again, I have to switch focus to another window or workspace and focus back to Firefox.

This happens under Arch Linux with Sway, both in Wayland and XWayland mode.

**To Reproduce**

1. Go to https://github.com/gdh1995/vimium-c.
2. Press `f` to enter hint mode.
3. Try pressing `Esc`. In nearly all cases, it doesn't do anything. (Although pressing any of the hint keys still work.)
4. Focus to another window and focus back to Firefox.
5. Pressing `Esc` works now.

**Browser, OS and Vimium C versions**

- Browser name: Firefox
- Browser version: 107.0
- Vimium C version: v1.99.9
- OS name and version: Arch Linux

**Comment by gdh1995** [comment]
Please test `Escape` and other keys on your keyboard on https://gdh1995.cn/vimium-c/keyboard-test.html .

**Comment by maxigaz** [comment]
> Please test `Escape` and other keys on your keyboard on https://gdh1995.cn/vimium-c/keyboard-test.html .

They all work on that page. Escape in particular works when the input box is not in focus. (When it is in focus, pressing Esc once removes focus from it.)

**Comment by gdh1995** [comment]
Does it logs some `keyCode`s of `0`?

**Comment by gdh1995** [comment]
Did it log some `keyCode`s of `0`?

**Comment by gdh1995** [comment]
Did it log some `keyCode`s of `0`?

**Comment by maxigaz** [comment]
> Did it log some `keyCode`s of `0`?

No, all the KeyCodes in the last column are numbers higher than zero.

**Comment by gdh1995** [comment]
Oh sorry it's a bug of the new fix which was to detect repeated keys.

**Comment by gdh1995** [comment]
Hello you may try this: [vimium_c-1.99.9-firefox-3a829c8-mod.zip](https://github.com/gdh1995/vimium-c/files/10056565/vimium_c-1.99.9-firefox-3a829c8-mod.zip) .

A newer version will be released in up to 24 hours.

**Comment by maxigaz** [comment]
> Hello you may try this: [vimium_c-1.99.9-firefox-3a829c8-mod.zip](https://github.com/gdh1995/vimium-c/files/10056565/vimium_c-1.99.9-firefox-3a829c8-mod.zip) .

It looks like you fixed it. Thank you!

> A newer version will be released in up to 24 hours.

Awesome! I'll keep this issue open in the meantime.

**Comment by xuesoso** [comment]
Where should the contents of ".zip" file be placed? Is it possible to disable automatic update of the add-on? Thank you!

**Comment by gdh1995** [comment]
Well, because I'm a bit busy today, I'll release v1.99.90 directly, but not wait for new bug reports. You may try this button:
![image](https://user-images.githubusercontent.com/5547703/203255543-a519ebe9-d471-4739-b5e4-26654352d47a.png)

**Comment by maxigaz** [comment]
I've just upgraded to v1.99.90. All seems to work fine. Thank you again!
